src/lsp/cobol_ptree/branching_statements.ml

```diff
@@ -21,52 +21,38 @@ open Simple_statements
 
 (* --- *)
 
-(* GOTO <target> *)
+(* GOTO *)
 type goto_stmt =
-  {
-    goto_target: procedure_name with_loc;
-  }
-[@@deriving ord]
-
-let pp_goto_stmt ppf { goto_target } =
-  Fmt.pf ppf "@[GO TO @[%a@]" pp_procedure_name' goto_target
-
-(* GOTO ENTRY *)
-type goto_entry =
-  {
-    goto_entry_target: alphanum_string
-  }
-[@@deriving ord]
-
-let pp_goto_entry ppf { goto_entry_target } =
-  Fmt.pf ppf "@[GO TO @[%a@]" pp_alphanum_string goto_entry_target
-
-(* GOTO DEPENDING *)
-type goto_depending_stmt =
-  {
-    goto_depending_targets: procedure_name with_loc nel;
-    goto_depending_on: ident;
-  }
+  | GoToSimple of
+    {
+      goto_targets: procedure_name with_loc nel;
+      goto_depending_on: ident option;
+    }
+  | GoToEntry of
+    {
+      goto_entry_targets: alphanum_string with_loc nel;
+      goto_entry_depending_on: ident option;
+    }
 [@@deriving ord]
 
-let pp_goto_depending_stmt ppf { goto_depending_targets; goto_depending_on } =
-  Fmt.pf ppf "@[GO TO @[%a@]@ DEPENDING ON %a"
-    (NEL.pp ~fsep:"@ " pp_procedure_name') goto_depending_targets
-    pp_ident goto_depending_on
-
-(* GOTO ENTRY DEPENDING *)
-type goto_entry_depending_stmt =
-  {
-    goto_entry_depending_targets: alphanum_string nel;
-    goto_entry_depending_on: ident;
-  }
-[@@deriving ord]
+let pp_goto_stmt ppf = function
+  | GoToSimple { goto_targets; goto_depending_on = None } ->
+    Fmt.pf ppf "@[GO TO @[%a@]"
```

Comment:
There's a missing `@]` at the end of every format string that starts with `@[GO TO`